lib/gui/pages/main/controllers/image-selection.js

```diff
@@ -65,14 +72,42 @@ module.exports = function(SupportedFormatsModel, SelectionStateModel, AnalyticsS
       return;
     }
 
-    SelectionStateModel.setImage(image);
+    if (!SupportedFormatsModel.looksLikeWindowsImage(image.path)) {
+      SelectionStateModel.setImage(image);
 
-    // An easy way so we can quickly identify if we're making use of
-    // certain features without printing pages of text to DevTools.
-    image.logo = Boolean(image.logo);
-    image.bmap = Boolean(image.bmap);
+      // An easy way so we can quickly identify if we're making use of
```

Comment:
We can unify this whole thing and avoid repeating the image selection logic with something like this:

```javascript
Bluebird.try(() => {
  if (!SupportedFormatsModel.looksLikeWindowsImage(image.path)) {
    return true;
  }

  AnalyticsService.logEvent('Possibly Windows image', image);

  // TODO: `Continue` should be on a red background (dangerous action) instead of `Change`.
  // We want `X` to act as `Continue`, that's why `Continue` is the `rejectionLabel`
  return WarningModalService.display({
    confirmationLabel: 'Change',
    rejectionLabel: 'Continue',
    description: messages.warning.looksLikeWindowsImage()
  });
}).then((shouldContinue) => {

  if (!shouldContinue) {
    return this.reselectImage();
  }

  SelectionStateModel.setImage(image);

  // An easy way so we can quickly identify if we're making use of
  // certain features without printing pages of text to DevTools.
  image.logo = Boolean(image.logo);
  image.bmap = Boolean(image.bmap);

  AnalyticsService.logEvent('Select image', image);
}).catch(ErrorService.reportException);
```

Also notice that I catch the whole promise error and report it, so we don't hide uncaught exceptions accidentally.